Cabal/Distribution/Parsec/Types/ParseResult.hs

```diff
@@ -0,0 +1,52 @@
+{-# LANGUAGE CPP #-}
+module Distribution.Parsec.Types.ParseResult where
+
+import Prelude ()
+import Distribution.Compat.Prelude
+
+import Distribution.Parsec.Types.FieldDescr
+import Distribution.Parsec.Types.Common (PError (..), PWarning (..), Position (..))
+import Distribution.Parsec.Class (ParsecBS)
+
+import qualified Data.ByteString as BS
+import qualified Text.Parsec as P
+
+-- | TODO: we could accumulate errors as well
+data ParseResult a = PR [PWarning] (Either PError a)
```

Comment:
I think the Writer monad stack leak is fairly severe and we should strictify it. See http://stackoverflow.com/questions/25749126/a-stricter-control-monad-trans-writer-strict